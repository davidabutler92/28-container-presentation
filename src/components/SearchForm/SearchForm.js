import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { fetchNewsArticle } from '../services/newsApi';

function SearchForm({ setArticles, setLoading }) {
  const [input, setInput] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    setLoading(true);
    fetchNewsArticle(input).then((articles) => {
      setArticles(articles);
      setLoading(false);
    });
  };

  const handleInputChange = ({ target }) => {
    setInput(target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Search"
        value={input}
        onChange={handleInputChange}
      ></input>
      <button type="submit">Search</button>
    </form>
  );
}

SearchForm.propTypes = {
  setArticles: PropTypes.func,
  setLoading: PropTypes.func,
};

export default SearchForm;
