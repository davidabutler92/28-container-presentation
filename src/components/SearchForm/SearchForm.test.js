import React from 'react';
import { render, cleanup } from '@testing-library/react';
import SearchForm from './SearchForm';

describe('SearchForm component', () => {
  afterEach(() => cleanup());
  it('it should render a search form', () => {
    const { asFragment } = render(<SearchForm />);
    expect(asFragment()).toMatchSnapshot();
  });
});
