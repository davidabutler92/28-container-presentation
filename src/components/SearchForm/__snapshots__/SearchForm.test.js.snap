// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm component it should render a search form 1`] = `
<DocumentFragment>
  <form>
    <input
      placeholder="Search"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search
    </button>
  </form>
</DocumentFragment>
`;
